### ISSUE TITLE
Need help: unable to perform merge on azure sql database

### ISSUE THREAD

**Issue by jonyeezs** [roadrunnerlenny/etlbox#44]
Unfortunately unable to perform the merge on azure sql database.

```
System.AggregateException: One or more errors occurred. (USE statement is not supported to switch between databases. Use a new connection to connect to a different database.) ---> System.Data.SqlClient.SqlException: USE statement is not supported to switch between databases. Use a new connection to connect to a different database.
   at System.Data.SqlClient.SqlConnection.OnError(SqlException exception, Boolean breakConnection, Action`1 wrapCloseInAction)
   at System.Data.SqlClient.SqlInternalConnection.OnError(SqlException exception, Boolean breakConnection, Action`1 wrapCloseInAction)
   at System.Data.SqlClient.TdsParser.ThrowExceptionAndWarning(TdsParserStateObject stateObj, Boolean callerHasConnectionLock, Boolean asyncClose)
   at System.Data.SqlClient.TdsParser.TryRun(RunBehavior runBehavior, SqlCommand cmdHandler, SqlDataReader dataStream, BulkCopySimpleResultSet bulkCopyHandler, TdsParserStateObject stateObj, Boolean& dataReady)
   at System.Data.SqlClient.SqlCommand.RunExecuteNonQueryTds(String methodName, Boolean async, Int32 timeout, Boolean asyncWrite)
   at System.Data.SqlClient.SqlCommand.InternalExecuteNonQuery(TaskCompletionSource`1 completion, Boolean sendToPipe, Int32 timeout, Boolean asyncWrite, String methodName)
   at System.Data.SqlClient.SqlCommand.ExecuteNonQuery()
   at ALE.ETLBox.ConnectionManager.SqlConnectionManager.BeforeBulkInsert()
   at ALE.ETLBox.ControlFlow.DbTask.BulkInsert(ITableData data, String tableName)
   at ALE.ETLBox.DataFlow.DBDestination`1.WriteBatch(TInput[] data)
   at ALE.ETLBox.DataFlow.DBDestination`1.<InitObjects>b__59_0(TInput[] d)
   at System.Threading.Tasks.Dataflow.ActionBlock`1.ProcessMessage(Action`1 action, KeyValuePair`2 messageWithId)
   at System.Threading.Tasks.Dataflow.ActionBlock`1.<>c__DisplayClass6_0.<.ctor>b__0(KeyValuePair`2 messageWithId)
   at System.Threading.Tasks.Dataflow.Internal.TargetCore`1.ProcessMessagesLoopCore()
   --- End of inner exception stack trace ---
   at System.Threading.Tasks.Task.Wait(Int32 millisecondsTimeout, CancellationToken cancellationToken)
   at ALE.ETLBox.DataFlow.DBDestination`1.Wait()
```

This is why: https://social.msdn.microsoft.com/Forums/en-US/1d83c593-095d-416b-b927-7a6489503131/sql-quotusequot-statement-is-not-supported-in-azure-sql-database-for-selecting-different?forum=ssdsgetstarted

Could there be another option instead of using `USE`?

**Comment by jonyeezs** [comment]
Here's is their way of interacting between databases
https://azure.microsoft.com/en-us/blog/querying-remote-databases-in-azure-sql-db/

**Comment by roadrunnerlenny** [comment]
Hi @jonyeezs ,
thanks for using ETLBox.  I assume you are using the `DBMerge` component? 
I am a little bit puzzled with the error message - actually, there shouldn't be any USE-statement involved when using the DBDestination or DBMerge (unless the ConnectionManager has the `ModifyDBSettings` set to true manually, but this is turned off by default). 

So I would like to reproduce the error in my test environment. 

To get the right setup here, could you provide me the code snippet that you are actually using to perform the Merge? Especially I am interested how you create the connection manager (or if you just pass the ConnectionString) and the destination part of your data flow?

**Comment by jonyeezs** [comment]
Hi sorry i left out a crucial information. I'm still using v1.5.0.
Unfortunately we can't upgrade to the latest just yet.

I'll just override the methods.

**Comment by roadrunnerlenny** [comment]
Hi @jonyeezs ,

thanks for the feedback. I just added two test cases to test project: reading/writing data into an Azure Sql databse and merging data using the DBMerge. I used an Azure Sql Database (database only, configured as "serverless"). I can verify that the current version of ETLBox (1.7.1) worked fine for these scenarios.

Let me know when you were able to upgrade to the latest version and still encounter errors.
